test/MinimalDelegation.4337.t.sol

```diff
@@ -92,14 +93,17 @@ contract MinimalDelegation4337Test is DelegationHandler, TokenHandler, ExecuteHa
     /// forge-config: ci.isolate = true
     function test_handleOps_single_P256_gas() public {
         TestKey memory p256Key = TestKeyManager.initDefault(KeyType.P256);
+        bytes32 keyHash = p256Key.toKeyHash();
 
         vm.prank(address(signerAccount));
         signerAccount.register(p256Key.toKey());
 
         Call[] memory calls = CallBuilder.init();
         calls = calls.push(buildTransferCall(address(tokenA), address(receiver), 1e18));
 
-        bytes memory opData = bytes("");
+        /// Pass keyHash in opData
+        /// This can be spoofed! Set it to 0 to bypass the permissions check set above
+        bytes memory opData = abi.encode(bytes32(0));
         bytes memory executionData = abi.encode(calls, opData);
```

Comment:
I feel like actually this should revert in the contract because its encoding execution data that is unexpected in the executeUserOp flow

Reply:
it means that our CalldataDecoder lib is not handling length checks, which actually I did know about & is a TODO, but if we were to decode in memory this would actually cause a revert